### PR TITLE
Allow multiple occurrences of a combinator as part of single route

### PR DESCRIPTION
**Pull request by jhrcek**
This changes the internal data structure used to accumulate information from the API specification.
Instead of `Ordered.Map Parameter Details` I'm switching back to `[(Parameter, Details)]`.
This fixes both https://github.com/Holmusk/servant-docs-simple/issues/17 and https://github.com/Holmusk/servant-docs-simple/issues/18

### CODE REVIEW

File: src/Servant/Docs/Simple/Render.hs
Comment (turboMaCk) [inline src/Servant/Docs/Simple/Render.hs:94]: nice catch there

File: test/Test/Servant/Docs/Simple/Samples.hs
Comment (turboMaCk) [inline test/Test/Servant/Docs/Simple/Samples.hs:123]: :+1: nice.

**Review by turboMaCk — APPROVED** [review]
LGTM but please wait for review from @oishikg before merge.

**Comment by jhrcek** [comment]
Thanks for review and sure, I'll wait for Oishik's review.
In the meantime I found out this change also fixes https://github.com/Holmusk/servant-docs-simple/issues/18 which turns out to have the same root cause.
There are 2 endpoints nested under the same path in meadow:
https://github.com/Holmusk/meadow/blob/12cb572327ff81c92ea3b0b40eb3d89409dfac8e/backend/cdmp-meadow/src/Meadow/Server/App/MoodAndJournaling.hs#L44-L52

So the reason why the PUT endpoint is not rendered is because both those endpoints end up under the same key "/api/meadow/mood-journaling/entry/{entry-id::Id MoodAndJournalEntryType}" in the map.

**Review by oishikg — APPROVED** [review]
LGTM! Thanks for making these changes.

**Pull request merged**
